packages/backend-storage/API.md

```diff
@@ -7,13 +7,36 @@
 import { AmplifyUserErrorOptions } from '@aws-amplify/platform-core';
 import { BackendOutputStorageStrategy } from '@aws-amplify/plugin-types';
 import { CfnBucket } from 'aws-cdk-lib/aws-s3';
+import { Construct } from 'constructs';
 import { ConstructFactory } from '@aws-amplify/plugin-types';
 import { ConstructFactoryGetInstanceProps } from '@aws-amplify/plugin-types';
+import { EventType } from 'aws-cdk-lib/aws-s3';
 import { FunctionResources } from '@aws-amplify/plugin-types';
 import { IBucket } from 'aws-cdk-lib/aws-s3';
+import { IFunction } from 'aws-cdk-lib/aws-lambda';
 import { ResourceAccessAcceptor } from '@aws-amplify/plugin-types';
 import { ResourceAccessAcceptorFactory } from '@aws-amplify/plugin-types';
 import { ResourceProvider } from '@aws-amplify/plugin-types';
+import { StorageBucketsPayload } from '@aws-amplify/backend-output-schemas';
+import { StorageOutput } from '@aws-amplify/backend-output-schemas';
+
+// @public
+export class AmplifyStorage extends Construct implements ResourceProvider<StorageResources> {
+    constructor(scope: Construct, id: string, props: AmplifyStorageProps);
+    addTrigger: (events: EventType[], handler: IFunction) => void;
+    // (undocumented)
+    readonly resources: StorageResources;
+}
+
+// @public
+export class AmplifyStorageFactory implements ConstructFactory<ResourceProvider<StorageResources>> {
+    constructor(props: AmplifyStorageFactoryProps, importStack?: string | undefined);
+    // (undocumented)
+    static factoryCounter: number;
+    getInstance: (getInstanceProps: ConstructFactoryGetInstanceProps) => AmplifyStorage;
+    // (undocumented)
+    static hasDefault: boolean;
+}

```

Comment:
Why are we exporting these types ?

Reply:
Because we need to use use `AmplifyStorageFactory.isDefault` and counter in packages/backend/src/backend_factory.ts

Reply:
Information about storage and buckets must not leak into `packages/backend/src/backend_factory.ts` . 
We'd be exposing implementation detail of vertical. We can't do that.
There should be some other way to achieve that that is scoped to `backend-storage`. 

Please check if you could either:
1. Hook up into validation callbacks of CDK constructs
2. Add CDK aspect to storage stack that inspects child nodes.

The goals are:
1. Throw error at synthesis
2. Do not leak storage information to core components.

Reply:
Done [50ebda5](https://github.com/aws-amplify/amplify-backend/pull/1742/commits/50ebda51550a3839e566bcbe4bb884e478d63c3d)

Used `Aspects` to validate.